snuba/clickhouse/http.py

```diff
@@ -46,44 +46,106 @@ def encode(self, value: WriterTableRow) -> JSONRow:
         )
 
 
+class InsertStatement:
+    def __init__(self, table_name: str) -> None:
+        self.__table_name = table_name
+        self.__database: Optional[str] = None
+        self.__format: Optional[str] = None
+        self.__column_names: Optional[Sequence[str]] = None
+
+    def with_database(self, database_name: str) -> InsertStatement:
+        self.__database = database_name
+        return self
+
+    def with_format(self, format: str) -> InsertStatement:
+        self.__format = format
+        return self
+
+    def with_columns(self, column_names: Sequence[str]) -> InsertStatement:
+        self.__column_names = column_names
+        return self
+
+    def get_qualified_table(self) -> str:
+        return (
+            f"{self.__database}.{self.__table_name}"
+            if self.__database
+            else self.__table_name
+        )
+
+    def build_statement(self) -> str:
+        columns_statement = (
```

Comment:
Isn't this an error if there are no columns?

Reply:
No. If you use JSON format clickhouse does not need the columns list. It picks them from each message. For CSV encoding instead (even with headers) it wants the columns.